microprot/snakemake/Snakefile

```diff
@@ -3,192 +3,276 @@
     log:
-        'results/log/{seq}.split_cm.log'
+        'results/log/{seq}/split_cm.log'
     run:
-        for inp in open(input[0], 'r').readlines():
-            mask_sequence(inp, output[0], config['split_CM']['params'])
+        split_x(input[0], output[0],
+                step="CM",
+                frag_len=config['split_CM']['params']['min_fragment_length'],
+                e_val=config['split_CM']['params']['max_evalue'], log=log[0])
+
 
 rule search_Pfam:
     input:
-        split='results/{seq}/05-split_CM/{seq}.split_cm',
-        seq='data/{seq}.fasta'
+        config['MICROPROT_MAIN']+'results/04-split_CM/{seq}/{seq}'
     output:
-        'results/{seq}/06-search_Pfam/{seq}'
+        config['MICROPROT_MAIN']+'results/05-search_Pfam/{seq}/{seq}'
     log:
         'results/log/{seq}/search_pfam.log'
     run:
-        shell_f = shell('{paths[TOOLS][hhsuite]}/bin/hhsearch -i {domain} \
-                         {config[search_Pfam][params]} \
-                         -d {paths[DBS][pfamA]} \
-                         -o {out} \
-                         -oa3m {a3m}')
-        search_X(input['seq'], input['split'], output[0], shell_f)
+        search_x(input[0], output[0],
+                 config=config['search_Pfam']['params'],
+                 dbs=paths['DBS']['pfamA'],
+                 log=log[0])
+
 
 rule split_Pfam:
     input:
-        'results/{seq}/06-search_Pfam/{seq}'
+        rules.search_Pfam.output[0]
     output:
-        'results/{seq}/07-split_Pfam/{seq}.split_pfam'
+        config['MICROPROT_MAIN']+'results/06-split_Pfam/{seq}/{seq}'
     log:
-        'results/log/{seq}.split_pfam.log'
+        'results/log/{seq}/split_pfam.log'
     run:
-        for inp in open(input[0], 'r').readlines():
-            mask_sequence(inp, output[0], config['split_Pfam']['params'])
+        split_x(input[0], output[0],
+                step="Pfam",
+                frag_len=config['split_Pfam']['params']['min_fragment_length'],
+                e_val=config['split_Pfam']['params']['max_evalue'],
+                prob=config['split_Pfam']['params']['min_prob'], log=log[0])
+
 
 rule MSA_hhblits:
     input:
-        split='results/{seq}/07-split_Pfam/{seq}.split_pfam',
-        seq='data/{seq}.fasta'
+        config['MICROPROT_MAIN']+'results/06-split_Pfam/{seq}/{seq}'
     output:
-        'results/{seq}/08-MSA_hhblits/{seq}'
+        config['MICROPROT_MAIN']+'results/07-MSA_hhblits/{seq}/{seq}'
     log:
-        'results/log/{seq}.msa_hhblits.log'
+        'results/log/{seq}/msa_hhblits.log'
     run:
-        shell_f = shell('{paths[TOOLS][hhsuite]}/bin/hhblits -i {domain} \
-                         {config[MSA_hhblits][params]} \
-                         -d {paths[DBS][hhblits_db]} \
-                         -o {out} \
-                         -oa3m {a3m}')
-        search_X(input['seq'], input['split'], output[0], shell_f)
+        """
+        it's essentially a copy of `search_x` function, but here we're using
```

Comment:
agreed, but then search_x needs to be rewritten, to take either hhblits or hhsearch - each of those programs has different commandline structure!